test/format/hs.c

```diff
@@ -0,0 +1,117 @@
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "format.h"
+
+/*
+ * hs_cursor --
+ *     Do history store cursor operations.
+ */
+WT_THREAD_RET
+hs_cursor(void *arg)
+{
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_ITEM hs_key, hs_value;
+    WT_SESSION *session;
+    wt_timestamp_t hs_durable_timestamp, hs_start_ts, hs_stop_durable_ts;
+    uint64_t hs_counter, hs_upd_type;
+    uint32_t hs_btree_id, i;
+    u_int period;
+    int exact;
+
+    (void)(arg); /* Unused parameter */
+
+    conn = g.wts_conn;
+
+    /*
+     * Trigger the internal WiredTiger cursor order checking on the history-store file. Open a
+     * cursor on the history-store file, retrieve some records, close cursor, repeat.
+     *
+     * Open a session.
+     */
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+
+    memset(&hs_key, 0, sizeof(hs_key));
+    memset(&hs_value, 0, sizeof(hs_value));
+    hs_start_ts = 0; /* [-Wconditional-uninitialized] */
+    hs_counter = 0;  /* [-Wconditional-uninitialized] */
+    hs_btree_id = 0;
+    for (;;) {
+        /*
+         * open_cursor can return EBUSY if concurrent with a metadata operation, retry in that case.
+         */
+        while ((ret = session->open_cursor(session, WT_HS_URI, NULL, NULL, &cursor)) == EBUSY)
+            __wt_yield();
+        testutil_check(ret);
+
+        /*
+         * The history file has mostly tombstones, ignore them and retrieve the underlying values.
+         * We don't care about tombstones, but we do want to hit every key rather than skip over
+         * them. This is a rollback-to-stable flag we're using for our own purposes.
+         */
+        F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
+
+        /* Search to the last-known location. */
```

Comment:
What happens if we hit the end of the history store? Isn't this going to keep searching for the last key, doing a next to get `WT_NOTFOUND`, rinse repeat?

Reply:
Yes, I think you're correct -- it's not guaranteed, but you're right most of the time, at the least. I've pushed a fix, would you please re-review, @tetsuo-cpp?